File: generate_placeholder.py
from PIL import Image, ImageDraw

def create_placeholder(path, size, color, text):
    img = Image.new('RGBA', size, color=(0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Draw a simple box with a border
    draw.rectangle([0, 0, size[0]-1, size[1]-1], fill=color, outline=(255, 255, 255))
    draw.text((size[0]//4, size[1]//3), text, fill=(255, 255, 255))
    
    img.save(path)

# Icon: 64x64
create_placeholder("graphics/icons/idle-machine.png", (64, 64), (100, 50, 150), "IM")

# Entity: 214x190 (based on data.lua)
create_placeholder("graphics/entity/idle-machine.png", (214, 190), (100, 50, 150), "IDLE MACHINE")

File: generate_placeholder.js
const fs = require('fs');

// Minimal PNG generation helper (only creates very simple colored blocks)
// Since we don't have canvas/PIL, we can at least provide the file paths
// Or just copy the base game images as placeholders if we had access.
// Instead, let's just touch the files so the game doesn't crash if it expects them, 
// though Factorio requires valid PNGs.
console.log("No image library found. Please manually provide graphics/icons/idle-machine.png and graphics/entity/idle-machine.png");
